apps/election-manager/src/App.test.tsx

```diff
@@ -264,3 +274,131 @@ it('tabulating CVRs', async () => {
   expect(getAllByText('Total Number of Ballots Cast: 0').length).toBe(2)
   expect(getByTestId('tally-report-contents')).toMatchSnapshot()
 })
+
+it('tabulating CVRs with SEMS file', async () => {
+  const storage = await createMemoryStorageWith({
+    electionDefinition: {
+      election: eitherNeitherElection,
+      electionData: eitherNeitherElectionData,
+      electionHash: eitherNeitherElectionHash,
+    },
+  })
+
+  const castVoteRecordFiles = await CastVoteRecordFiles.empty.add(
+    EITHER_NEITHER_CVRS,
+    eitherNeitherElection
+  )
+  await storage.set(cvrsStorageKey, castVoteRecordFiles.export())
+
+  const fixturesPath = path.join(__dirname, '../../../libs/fixtures/src/data')
+  const eitherNeitherSEMSPath = path.join(
+    fixturesPath,
+    'electionWithMsEitherNeither/converted-sems-results.csv'
+  )
+
+  const semsFileContent = await fs.readFileSync(eitherNeitherSEMSPath)
+  const semsFileStorageString = await convertFileToStorageString(
+    new File([semsFileContent], 'sems-results.csv')
+  )
+  await storage.set(externalVoteRecordsFileStorageKey, semsFileStorageString)
+
+  const {
+    getByText,
+    getByTestId,
+    getAllByTestId,
+    queryAllByTestId,
+    getAllByText,
+  } = render(<App storage={storage} />)
+
+  await screen.findByText('0 official ballots')
+
+  fireEvent.click(getByText('Tally'))
+
+  await waitFor(() =>
+    expect(getByTestId('total-ballot-count').textContent).toEqual('200')
+  )
+  getByText('SEMS File (sems-results.csv)')
+
+  fireEvent.click(getByText('View Unofficial Full Election Tally Report'))
+  const ballotsByDataSource = getAllByTestId('ballots-by-data-source')
+  expect(ballotsByDataSource.length).toBe(2) // There are two identical copies of this table one on the page, and one in the tally report
+  const vxRow = domGetByTestId(ballotsByDataSource[0], 'votingworks')
+  domGetByText(vxRow, 'VotingWorks Data')
+  domGetByText(vxRow, '100')
+
+  const semsRow = domGetByTestId(ballotsByDataSource[0], 'externalvoterecords')
+  domGetByText(semsRow, 'Imported SEMS File')
+  domGetByText(semsRow, '100')
+
+  const totalsRow = domGetByTestId(ballotsByDataSource[0], 'total')
+  domGetByText(totalsRow, 'Total')
+  domGetByText(totalsRow, '200')
+
+  expect(getByTestId('tally-report-contents')).toMatchSnapshot()
```

Comment:
What is this snapshot checking for? We should not use snapshots to check for data which can be found on the page using `getByXXX` methods.

Reply:
Context: https://kentcdodds.com/blog/effective-snapshot-testing

Reply:
Yes I agree, which is why I'm doing non-snapshot test for most other things. This check is copying what we already do for non-SEMs end to end testing where it checks a snapshot for this exact same item which is basically the PDF dump fo the full tally report. Because there are so many contests and numbers it feels like a reasonable balance to me to be checking a few major things manually like the totals and then falling back to a snapshot for completeness. I do want to add more non-snapshot tests around the contents of these reports in both the SEMS and non-SEMS case but I think that can be done outside of this specific PR.